File: solutions/javascript/nth-prime/1/nth-prime.js
export const prime = (n) => {
  if (n === 0) throw new Error('there is no zeroth prime');
  const primes = [];
  let i = 2;
  while (primes.length < n) {
    let isPrime = true;
    for (const p of primes) {
      if (p * p > i) break;
      if (i % p === 0) {
        isPrime = false;
        break;
      }
    }
    if (isPrime) primes.push(i);
    i++;
  }
  return primes[primes.length - 1];
};

File: solutions/ruby/nth-prime/1/nth_prime.rb
class Prime
  def self.nth(n)
    raise ArgumentError if n < 1
    primes = []
    candidate = 2
    while primes.length < n
      primes << candidate if primes.all? { |p| candidate % p != 0 }
      candidate += 1
    end
    primes.last
  end
end
